bugbot/rules/webcompat_score.py

```diff
@@ -0,0 +1,88 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from typing import Any, Optional
+
+from bugbot import gcp
+from bugbot.bzcleaner import BzCleaner
+
+
+class WebcompatScore(BzCleaner):
+    def __init__(self):
+        super().__init__()
+        self.autofix_changes = {}
+        self.scored_bugs = {}
+
+    def description(self) -> str:
+        return "Update WebCompat score fields"
+
+    def filter_no_nag_keyword(self) -> bool:
+        return False
+
+    def has_default_products(self) -> bool:
+        return False
+
+    def get_autofix_change(self) -> dict[str, Any]:
+        return self.autofix_changes
+
+    def handle_bug(
+        self, bug: dict[str, Any], data: dict[str, Any]
+    ) -> Optional[dict[str, Any]]:
+        bug_id = bug["id"]
+
+        if (
+            bug_id in self.scored_bugs
+            and bug["cf_webcompat_score"] != self.scored_bugs[bug_id]
+        ):
+            self.autofix_changes[bug_id] = {
+                "cf_webcompat_score": self.scored_bugs[bug_id]
+            }
+            return bug
+
+        return None
+
+    def get_bz_params(self, date) -> dict[str, Any]:
+        fields = ["id", "cf_webcompat_score"]
+        self.scored_bugs = self.get_bug_data()
+        return {
+            "include_fields": fields,
+            "classification": "Client Software",
+            "resolution": "---",
+            "j_top": "OR",
+            "f1": "OP",
+            "f2": "product",
+            "o2": "equals",
+            "v2": "Web Compatibility",
+            "f3": "component",
+            "o3": "equals",
+            "v3": "Site Reports",
+            "f4": "CP",
+            "f5": "OP",
+            "f6": "product",
+            "o6": "notequals",
+            "v6": "Web Compatibility",
+            "f7": "keywords",
+            "o7": "substring",
+            "v7": "webcompat:site-report",
+            "f8": "CP",
+        }
+
+    def get_bug_data(self) -> dict[int, str]:
```

Comment:
I would use a more descriptive method name, e.g.,:
```suggestion
    def fetch_bug_scores(self) -> dict[int, str]:
```